### DISCUSSION TITLE
Heatmeters ULTRAHEAT T230/T330

### DISCUSSION THREAD

**Discussion by shyrokorad** [Zeppelin500/MBusino#22] [General]
Was the operation of receiving telegrams from ULTRAHEAT T230/T330 checked?
I have two meters, T230 and T330. When connected, no data is received via M-Bus.
In MQTT, there is a message: MBUSerror = no_good_telegram.
Setting
![mbus2](https://github.com/user-attachments/assets/d62c7749-469a-472f-b186-a59a48a8ff75)
A printed circuit board was manufactured based on the provided schematic in the project. Additionally, a board was purchased on AliExpress.
![mbus](https://github.com/user-attachments/assets/08e67daf-eeb9-413d-ad0a-bd8519efa88d)
I tried connecting through a USB-to-MBus converter and using the MBSheet software. I was able to retrieve data from the meters.

**Comment by Zeppelin500** [comment]
Hello,
There are some not working chinese clones of the Master around.
Is the Photo yours? The PCBs with the small arrows are the good ones. See the [discussion](https://github.com/Zeppelin500/MBusino/discussions/18).

Install the standard MBusino (not 3S) firmware and activate M-Bus debug. If there ist a telegram, send it to me or test it at [Wokwi](https://wokwi.com/projects/402235052803622913) . If not, you have a Hardware issue.

**Comment by Zeppelin500** [comment]
PS: If you already received a telegram by the USB device. Show the raw Telegram here and I will confirm the capability of the MBusinoLib library or I will fix the Library, If decoding fails. But I think the Library is not the problem. We will see.

**Comment by shyrokorad** [comment]
Hello. I purchased a board via the link; it has triangles.
![photo_2024-12-24_15-31-39](https://github.com/user-attachments/assets/3c54e0bc-b157-4c59-8078-7897b89906a9)
My MBUS input is 28V, and when the D1 Mini starts polling, I can hear the high-frequency transformer working; there is a slight high-pitched sound.
Connection Diagram
[MBusinoLib.pdf](https://github.com/user-attachments/files/18240312/MBusinoLib.pdf)

The MBusino.ino.bin (0.9.11) file was uploaded.
Data is being sent via MQTT-Explorer.
![mbus4](https://github.com/user-attachments/assets/09a931ad-1543-4509-869f-fa3c4e61eadc)

 *If you already received a telegram by the USB device*
 I’m not very experienced with Arduino, so I need to figure out how to collect logs via USB.
In Arduino IDE, the logs show as squares.

**Comment by Zeppelin500** [comment]
You should have the good master.
As you showed, there is nothing received. --> nothing to decode.
Try to connect just GND, 5V, 3.3V, and Rx/TX with jumper wires. You do not need any capacitor.
Attention: the labeling of Tx/Rx at the master is not common. You have to connect Rx with Rx...

I can't help with your USB device. I do not own one.

**Comment by shyrokorad** [comment]
I assembled the circuit on a breadboard with direct wire connections: 3.3V, 5V, GND, RX-RX, TX-TX, and everything worked perfectly! 😉
![mbus5](https://github.com/user-attachments/assets/ba4aed97-453f-4e76-b412-1cc89adc2518)
telegram = 68E2E26808007232500969A73207040A1000000974040970080C06000000000C14000000000B2D0000000B3B0000000B5A1702000B5E9901000B621900000C7832500969891071073C22827806000C22827806000C26000000008C901006000000009B102D0000009B103B0000009B105A8303009B105E0605009410AD6F000000009410BB6F000000009410DA6F060F7C289410DE6F230E3C234C06000000004C14000000007C22679205004C2600000000CC90100600000000DB102D000000DB103B000000DB105A830300DB105E060500848F0F6D0000E1F1046D1717183C0F1407006201201600
That means I'll focus on finding the issue on the board. Thank you very much for the advice and help! 😊
I checked two meters, and the data is coming through.

**Reply by Zeppelin500** [reply]
I'm pleased.
Merry Christmas

**Comment by shyrokorad** [comment]
I removed two capacitors at the M-Bus input, and everything started working. I had found a connection diagram from the manufacturer on AliExpress, which recommended adding two capacitors.  Thank you so much for this project! Your efforts in creating it helped me solve an old problem: reading data from a meter located 2,000 km away. I used to have to ask friends for help.  Merry Christmas! 🎄😊
![mbus6](https://github.com/user-attachments/assets/e020f87c-42fe-411b-992e-719d0a8447ae)

**Comment by Zeppelin500** [comment]
Good to know.
Don't forget the Stars for MBusino and the Library :-)

**Comment by Zeppelin500** [comment]
Addendum:
The chinese diagram shows 1nF, not 1uF! That is 1000 times smaller as your capacitor. If you use the size you had written.
![Screenshot_2024-12-26-08-26-41-347_com alibaba aliexpresshd-edit](https://github.com/user-attachments/assets/404651f0-8999-4a72-a529-d5b266eef818)

**Comment by shyrokorad** [comment]
Yes. That was my mistake. I ordered the components incorrectly. It's working without capacitors now.
![mbus](https://github.com/user-attachments/assets/bc5e4aa6-5580-4788-87b6-0d83ddccecee)
Regarding the presence of capacitors at the input, I found information from the seller.
(https://www.aliexpress.com/item/1005006131052725.html?sourceType=1&spm=a2g0o.wish-manage-home.0.0)
I didn't order them.

**Comment by shyrokorad** [comment]
When polling devices powered by batteries, as I understand, it reduces the battery life, and it is recommended to increase the polling intervals. For instance, if there is no need to constantly poll the device, maybe there's an option to add a command via MQTT to retrieve parameters from M-Bus?
![Screenshot 2024-12-31 184243](https://github.com/user-attachments/assets/cf805c4e-046a-4919-b8b0-bfda89d36c54)

Happy New Year! 🎉 May this year bring you lots of happiness, health, and success! 🥳

**Comment by Zeppelin500** [comment]
Thank you! Happy New Year! 🎉
M-Bus is designed to supply the slaves with power. So, if the slave manufacturer follows the Meterbus idea. No need to reduce the polling interval to save battery life.
But not all manufacturer has implemented this correctly.
Is your Idea to change parameters via MQTT or just poll via MQTT? Both is implementable.

**Reply by shyrokorad** [reply]
For example, if there is no need to collect statistics from the heat meter, then constant polling is unnecessary.

Here’s my situation: I need to retrieve data from the meter once a month to submit the readings to the service provider. At other times, I just want to check the current data.

Currently, I’ve temporarily implemented automation using a smart plug that turns on for 80 seconds. I connected a D1 mini and collect readings from the meter once a day, saving them to an SQL database for daily chart statistics. If I need to check current data, I turn on the smart plug, and the current readings are sent to me via a Telegram bot.

For example, my idea is to have the device always on, with temperature sensors (DS18B20) being polled at an interval set during the initial MBusino configuration. However, the MBus device remains in "sleep mode" without being polled. Using MQTT, we can send a command like MBus_read = 1 to start polling the MBus device. Once the data is retrieved, the device goes back to "sleep mode" and waits for the next polling command.

**Reply by Zeppelin500** [reply]
OK, I will try to implement MQTT polling. Give me a couple of days.

**Reply by shyrokorad** [reply]
Thank you for your prompt response.
If you have time, please take a look at it. I don't need it urgently, as the board is already in another country, connected to a meter, and I can only update the firmware via web update.
I’ve ordered a similar board from AliExpress, where I bought the first one. I'm waiting for the new board to arrive, and I will continue testing and debugging once it’s here.
There are also new boards available now, which are said to support up to 200 devices.
https://www.aliexpress.com/item/1005008051612400.html?sourceType=1&spm=a2g0o.wish-manage-home.0.0

I also noticed an issue. I’m using the D1 Mini Pro board. When I use it, no data is received from the temperature sensor—there’s no data in MQTT. 
![Screenshot 2025-01-03 112408](https://github.com/user-attachments/assets/19157d8a-921c-4968-8502-709d0da2d3bd)

However, with the regular D1 Mini board, temperature readings are received correctly.
At the moment, I don’t need to use temperature data.

**Comment by Zeppelin500** [comment]
If you need to read out 200 slaves, this is the wrong solution. wmbusmeters will help you. :-)
I recommend the Zihatec master, but you have to use a ESP32 S2 mini. Which country do you come from?
Long time past since I tried the D1 mini pro. There were no issues. Do you use the 4M version? I have to search for my pro to test it.

By the way, [here](https://github.com/Zeppelin500/MBusino/tree/main/playground/MBusinoPolling) is the polling code.
Publish MQTT topic "MBusino/mbusPolling", (payload don't matter) and the slave will answer.
E.g. M-Bus Interval 2.592.000 sec. (30 days). If you need a records between, just send a MQTT message.

**Reply by shyrokorad** [reply]
**_If you need to read out 200 slaves, this is the wrong solution. wmbusmeters will help you. :-)_**
I don’t have that many devices :)
I’ve considered using wmbusmeters, but there are issues that prevent me from using it.

**_I recommend the Zihatec master, but you have to use a ESP32 S2 mini. Which country do you come from?_**
Ukraine. I have been living in Europe for 3 years, and my apartment is in Kharkiv. A heat meter is installed there, and during the heating season, I need to take readings and report them to the utility company. The meter has an MBus module, which I haven’t used for 4 years. I’m a Siemens PLC programmer, and DIY is my hobby. That’s why I bought 2 meters on ebay.de to solve the issue of remote data reading, as I don’t have physical access to my meter.
I found many different projects on the internet, but your project seemed easy to implement and set up without requiring professional knowledge. I configured two boards: a D1 Mini and a D1 Mini Pro (which has an external WiFi antenna).
When I tried to remotely change the MBus polling frequency, after rebooting, the D1 Mini couldn’t connect to the WiFi network. I replaced it with the D1 Mini Pro (apparently with 16MB), and everything started working, but the connected sensor didn’t show up in MQTT.

**_Long time past since I tried the D1 mini pro. There were no issues. Do you use the 4M version?_**
I also have a D1 Pro with 4MB available. I will check your updates today.
In the meantime, I’m waiting for the delivery of another board.
![mbus](https://github.com/user-attachments/assets/08e67daf-eeb9-413d-ad0a-bd8519efa88d)

**Comment by Zeppelin500** [comment]
Please send a mail to MBusinoLib@gmail. May be I can help you with hardware.

**Comment by shyrokorad** [comment]
A web update was successfully performed with the new MBusinoPolling firmware for the D1 Mini Pro, and no issues were found. The MQTT topic request "MBusino/mbusPolling" for retrieving MBus data works correctly.
Thank you for your hard work!
